Introduce metrics for broadcast transactions

thomaseizinger opened the pull request:
Together with a query powered by `increase()`, we can accurately measure
these counters even across restarts of the service (which resets the
counter to 0).

See https://github.com/siimon/prom-client/issues/364.

thomaseizinger committed:
Introduce metrics for broadcast transactions

Even though this counter resets upon application restarts, it still
allows us to measure how many transactions of each kind are broadcast.
With a prometheus query that utilises `rate` or `increase` we can even
somewhat handle the resets to 0 due to restarts and get an idea of how
many of these transactions are broadcast.

For example, we can create the following query:

```
increase(blockchain_transactions_broadcast_total{kind="commit"}[3d])
```

This query tells us the _increase_ of commit transactions each point in
time over the last 3 days.

luckysori commented on daemon/src/monitor.rs:
```diff
@@ -1000,6 +1018,48 @@ impl Actor {
+    prometheus::IntCounter,
+> = conquer_once::Lazy::new(|| {
+    prometheus::register_int_counter!(
+        "blockchain_collaborative_close_transactions_broadcast_total",
```
❓ Is the `blockchain_` prefix necessary?

thomaseizinger replied:
You are meant to have a "namespace" prefix and I couldn't come up with a better one. Inputs welcome!

luckysori commented on daemon/src/monitor.rs:
```diff
@@ -1000,6 +1018,48 @@ impl Actor {
+        )
+        .unwrap()
+    });
+static COLLABORATIVE_CLOSE_TRANSACTION_BROADCAST_COUNTER: conquer_once::Lazy<
```
❓ Wasn't it collaborative settlement? Nitty, but it would be good to use consistent terminology.

thomaseizinger replied:
Uhm, this file calls it collaborative close :sweat_smile:

bonomat reviewed (commented):
Why do you think this metric is useful?

thomaseizinger commented on daemon/src/monitor.rs:
```diff
@@ -1000,6 +1018,48 @@ impl Actor {
     }
 }
 
+static LOCK_TRANSACTION_BROADCAST_COUNTER: conquer_once::Lazy<prometheus::IntCounter> =
```
This should probably use an `IntCounterVec`.

thomaseizinger committed:
Replace `fmt::Display` with more efficient `name` fn

Calling `fmt::Display` via `ToString` involves an allocation which is
unnecessary if we always produce the same static string. Additionally,
we can make the logs friendlier to parse for downstream systems through
the usage of fields.

Removing `fmt::Display` in favor of a `name() -> &'static str` function
gives us both.

bonomat reviewed (approved):
Ok, I can think of how this is useful: 
REcentely I was wondering why so many positions were closed. With a look at this metric it would have been obvious that they were collab-closed and all is good without having to check the logs.

da-kami reviewed (approved):
Thanks! I am very much in favor of any metrics that can give us the status without having to go through the logs to find out what's going on.
